[Jeremy Yeo Zhi Chen] iP

rgbpokka opened the pull request:
# Duke - *The ToDoList you need*
> Let us do the tracking

Duke
* is text-based
* has no learning curve 
* ~~convenient~~ extremely convenient to use

Setup
1. Download from [here](https://github.com/nus-cs2103-AY2122S1/ip/pull/321)
2. The rest is self explanatory!

It is also **FREE** of charge! :smile: 

Features 
- [x] managing tasks
- [ ] managing deadlines
- [ ] managing events

We let you practise programming too! `let isTired = true`

```javascript
if (isTired) {
  return true
}
```

Commit:
Add Gradle support

rgbpokka committed:
Greet, Echo, Exit

rgbpokka committed:
Created ToDoList with ability to add and display ToDos

rgbpokka committed:
Added ability to mark tasks as done

rgbpokka committed:
Added support for tracking ToDos, Events and Deadlines

rgbpokka committed:
added automated text ui testing

rgbpokka committed:
added basic error handling

rgbpokka committed:
added support for deleting tasks from list

rgbpokka committed:
Add ability to save tasks in hard disk

rgbpokka committed:
Fix file write bug

rgbpokka committed:
Add ability to understand dates

rgbpokka committed:
Merge branch 'branch-Level-7'

rgbpokka committed:
Merge branch 'branch-Level-8'

rgbpokka committed:
Fix greet message typo

rgbpokka committed:
Refactor code to use more OOP

rgbpokka committed:
Put classes into Duke package

rgbpokka committed:
Add JUnit tests

rgbpokka committed:
Add JavaDoc comments

rgbpokka committed:
Tweak code to comply with coding standard

rgbpokka committed:
Add ability to find task by searching for keyword

rgbpokka committed:
Merge tag 'A-CodingStandard'

rgbpokka committed:
Merge tag 'Level-9'

rgbpokka committed:
Package app as JAR file

rgbpokka committed:
Merge remote-tracking branch 'origin/add-gradle-support' into A-Gradle

rgbpokka committed:
Set up gradle to build and run Duke

rgbpokka committed:
Merge branch A-Gradle

lwj1711 commented on data/duke.txt:
```diff
@@ -0,0 +1,5 @@
+E|1|project meeting|2021-12-08
```
I'm not entirely sure but I believe we do not need to commit the data files.

lwj1711 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
```
Javadocs for the class too will be good to have!

over-fitted replied:
Agree with @lwj1711

lwj1711 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    protected String by;
```
The naming 'by' might be a little ambiguous. Perhaps 'timeBy' might be better since it is a time.

lwj1711 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+
+    /**
+     *
+     * @param description
```
Do remember to flesh out the description for the parameters.

over-fitted replied:
In addition to fleshing out description, do consider removing unnecessary newlines in JavaDoc!

lwj1711 reviewed (commented):
Overall nice code quality, minor nit would be that javadocs could be a bit more consistent.

over-fitted commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+    /**
+     *
+     * @return
+     */
```
In addition to fleshing out description, do consider removing unnecessary newlines in JavaDoc!

Also, should the return value be explained here?

over-fitted commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+    /**
+     *
+     * @return
+     */
```
In addition to fleshing out description, do consider removing unnecessary newlines in JavaDoc!

Also, should the return value be explained here?

over-fitted commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+     *
+     * @return
+     */
+    @Override
+    public char getTaskType() { return 'D'; }
```
Since this is not a getter, it may be worthwhile to add javadocs for this method!

over-fitted commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+    /**
+     *
+     * @return
+     */
```
In addition to fleshing out description, do consider removing unnecessary newlines in JavaDoc!

Also, should the return value be explained here?

over-fitted commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+    private Storage storage;
+    private Parser parser;
+
+    /**
+     *
+     * @param filePath
+     */
```
In addition to fleshing out description, do consider removing unnecessary newlines in JavaDoc!

over-fitted commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+
+    /**
+     *
+     */
```
Should this comment be empty?

over-fitted commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,61 @@
+        }
+    }
+
+    /**
+     *
+     * @param args
+     */
```
In addition to fleshing out description, do consider removing unnecessary newlines in JavaDoc!

over-fitted commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,50 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Event extends Task {
```
Do add header comments for your classes!

over-fitted reviewed (commented):
I like how your functional tests are quite comprehensive, and that most of your classes follow proper style guidelines!

I think that your PR almost LGTM, after the remaining files have been revised to improve minor style details. :tada:

rgbpokka committed:
Create basic GUI

rgbpokka committed:
Remove debugging statement from Storage.java

rgbpokka committed:
Refactor code so Duke responds appropriately to user

rgbpokka committed:
Improve styling of chatbot

rgbpokka committed:
Fill up incomplete javadoc comments

rgbpokka committed:
Fix merge conflict in duke.txt

rgbpokka committed:
Add assertions to the code

Assertions are not used in the code.

Assertions are helpful in determining whether assumptions hold true and if mistakes have been made in the code.

Let's add some assertions to the deleteItem and getTask method in TaskList class. We will assert that the indices provided to these methods are within bounds.

The ability to detect if the index of the item to be retrieved/deleted is out of bounds is important as it suggests that the arguments passed to the methods are incorrect.

rgbpokka committed:
Improve code quality

The current code does not adhere to the coding quality standards.

The Single Layer of Abstraction (SLAP) rule is violated and there is nested code. Moreover, some class names are inappropriate considering their purpose.

Let's rename the UserInterface class as ResponseGenerator, as the purpose of the class is to generate String responses based on the user's input. In addition, let's reorganize the code in the Parser class to reduce nesting.

Reduced nesting of code and appropriate class naming will allow readers to better understand the code, and also allow us to better maintain the code.

rgbpokka committed:
Use Streams API

rgbpokka committed:
Add ability to deal with duplicate tasks

rgbpokka committed:
Merge pull request #1 from rgbpokka/branch-A-Assertions

Add assertions to the code

rgbpokka committed:
Merge branch 'master' into branch-A-CodeQuality

rgbpokka committed:
Merge pull request #2 from rgbpokka/branch-A-CodeQuality

Improve code quality

rgbpokka committed:
Merge branch 'master' into branch-A-Streams

rgbpokka committed:
Merge pull request #3 from rgbpokka/branch-A-Streams

Use Streams API

rgbpokka committed:
Fix merge conflict with branch-C-DetectDuplicates

rgbpokka committed:
Improve GUI

rgbpokka committed:
Add user guide
